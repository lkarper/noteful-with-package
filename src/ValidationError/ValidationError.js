import React from 'react';
import PropTypes from 'prop-types';

const ValidationError = (props) => {
    if (props.message) {
        return <div role="alert" className="validation-error">{props.message}</div>;
    }
    return <></>;
}

ValidationError.defaultProps = {
    message: '',
}

ValidationError.propTypes = {
    message: PropTypes.string,
}

export default ValidationError;