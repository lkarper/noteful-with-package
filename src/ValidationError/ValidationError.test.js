import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ValidationError from './ValidationError';

describe('ValidationError component', () => {
    
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ValidationError />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected when there is no error', () => {
        const wrapper = shallow(<ValidationError />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('renders the UI as expected when passed an error message in props', () => {
        const wrapper = shallow(<ValidationError message={"Failure to fetch"}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
    
});