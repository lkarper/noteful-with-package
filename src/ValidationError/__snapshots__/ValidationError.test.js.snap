// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationError component renders the UI as expected when passed an error message in props 1`] = `
<div
  className="validation-error"
>
  Failure to fetch
</div>
`;

exports[`ValidationError component renders the UI as expected when there is no error 1`] = `<Fragment />`;
